week_3_alt_assignment/index.js

```diff
@@ -0,0 +1,384 @@
+      whitespace.
+
+      And, yes, this is HTML inside JavaScript!
+      Inception!!: http://localhost:3000/static/inception_meme.jpg
+    --!>
+
+    <style>
+      /**
+       * And now some CSS
+       * in our HTML
+       * which is in our JavaScript
+       * :)
+       */
+
+      * { font-family: Arial; box-sizing: border-box; }
+      body { display: flex; align-items: center; justify-content: center; }
+      main { max-width: 70ch; }
+    </style>
+
+    <body><main>
+      <h1>Hello world!</h1>
+      <p>
+        Text is cool and all, but with <b><i>HTML</i></b> we can do all kinds
+        of fancy stuff!
+      </p>
+
+      <details>
+        <summary>For example, with nothing more than a little HTML, we can hide
+        a little surprise for you to find!</summary>
+
+        <img width="500px" src="/static/inception_meme.jpg" />
+
+        <p>
+          Turns out that serving up a little HTML and CSS can create a pretty
+          cool user experience!
+        </p>
+
+        <p>
+          And, it turns out that a server can do a whole lot more than just
+          share some hard-coded HTML. Before we check that out, there is one more
+          cool thing that HTML can do: we don't have to tell you the next link
+          to visit like we did before. Instead, we can hide the link behind some
+          text, and
+            <a href="/whatisaserver/dynamicHtml">all you need to do is click!</a>
+        </p>
+      </detail>
+
+    </main></body>
+
+  `);
+});
+
+// Now, I'm going to refactor the CSS into a separate string so that I can
+// re-use it for the remaining pages without repeating it! Instead you'll see
+// ${STYLE} at the beginning of each of the following HTML pages.
+const STYLE = `
+<style>
+  * { font-family: Arial; box-sizing: border-box; }
+  body { display: flex; align-items: center; justify-content: center; }
+  main { max-width: 70ch; }
+  pre { font-family: monospace; }
+</style>`;
+
+app.get("/whatisaserver/dynamicHtml", (request, response) => {
+  response.send(`
+    ${STYLE}
+    <body><main>
+    <p>
+      The most powerful thing about a server is that you can serve dynamic
+      content to the user. For example, I can actually inspect the request you
+      sent me, the server, to see where you're visiting from. For example, I
+      know all this about the browser you are sending a request from now:
+
+    <pre>${request.headers["user-agent"]}</pre>
+
+    <p>
+      Have you ever seen a URL with a bunch of jibberish in it? For example:
+    </p>
+    <pre>http://example.com?filter=1%2C56%2C3%2C7&sortOrder=id%3Adesc</pre>
+    <p>
+      Those are called URL parameters! It might look like gibberish, but
+      those are actually key-value pairs that the server can read while it
+      is processing a request. It's almost like using the URL as a tiny little
+      database; the URL itself can store data!
+    </p>
```

Comment:
```suggestion
    <p>
      The most powerful thing about a server is that you can serve dynamic
      content to the user. For example, I can actually inspect the request you
      sent me, the server, to see where you're visiting from. For example, I
      know all this about the browser you are sending a request from now:

    <pre>${request.headers["user-agent"]}</pre>

    <p>
      This is how the browser and the server can communicate with each other
      in order make your website dynamic. The server can decide to return different
      data based on information the browser gives it in the request.
    </p>
    <p>  
      Even the URL itself can have information that lets us change what we show
      on the website! Have you ever seen a URL with a bunch of gibberish in it? For example:
    </p>
    <pre>http://example.com?filter=1%2C56%2C3%2C7&sortOrder=id%3Adesc</pre>
    <p>
      Those are called URL parameters! It might look like gibberish, but
      those are actually key-value pairs that the server can read while it
      is processing a request. It's almost like using the URL as a tiny little
      database; the URL itself can store data!
    </p>
```